spec/models/signup_spec.rb

```diff
@@ -0,0 +1,105 @@
+require 'rails_helper'
+
+describe Signup do
+  before do
+    @params = { user: {first_name: "Joe",
```

Comment:
Those will be overriden by params from "invalid" context. This should go under the context.